tests/gis_tests/geoapp/test_functions.py

```diff
@@ -845,3 +862,48 @@ def test_argument_validation(self):
             City.objects.annotate(union=functions.GeoFunc(1, "point")).get(
                 name="Dallas"
             )
+
+
+class GeometryTypeFunctionTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.features = [
+            Feature.objects.create(name="Point", geom=Point(0, 0)),
+            Feature.objects.create(name="LineString", geom=LineString((0, 0), (1, 1))),
+            Feature.objects.create(
+                name="Polygon", geom=Polygon(((0, 0), (1, 0), (1, 1), (0, 0)))
+            ),
+            Feature.objects.create(
+                name="MultiPoint", geom=MultiPoint(Point(0, 0), Point(1, 1))
+            ),
+            Feature.objects.create(
+                name="MultiLineString",
+                geom=MultiLineString(
+                    LineString((0, 0), (1, 1)), LineString((1, 1), (2, 2))
+                ),
+            ),
+            Feature.objects.create(
+                name="MultiPolygon",
+                geom=MultiPolygon(
+                    Polygon(((0, 0), (1, 0), (1, 1), (0, 0))),
+                    Polygon(((1, 1), (2, 1), (2, 2), (1, 1))),
+                ),
+            ),
+        ]
+
+    @skipUnlessAnyDBFeature("has_GeometryType_function", "has_SDO_GTYPE_function")
```

Comment:
The entire class should be decorated with this
```suggestion
    @skipUnlessDBFeature("has_GeometryType_function")
```